packages/page-claims/src/Claim.tsx

```diff
@@ -3,24 +3,44 @@
 // of the Apache-2.0 license. See the LICENSE file for details.
 
 import { Option } from '@polkadot/types';
-import { BalanceOf, EthereumAddress } from '@polkadot/types/interfaces';
+import { BalanceOf, EthereumAddress, StatementKind } from '@polkadot/types/interfaces';
 
 import React, { useCallback, useEffect, useState } from 'react';
 import styled from 'styled-components';
-import { Button, Card } from '@polkadot/react-components';
+import { Button, Card, TxButton } from '@polkadot/react-components';
 import { useApi } from '@polkadot/react-hooks';
 import { FormatBalance } from '@polkadot/react-query';
 
 import { useTranslation } from './translate';
-import { addrToChecksum } from './util';
+import { addrToChecksum, getStatement } from './util';
 
 interface Props {
-  button: React.ReactNode;
+  accountId: string;
   className?: string;
   ethereumAddress: EthereumAddress | null;
+  ethereumSignature: string | null;
+  // Do we sign with `claims.claimAttest` (new) instead of `claims.claim` (old)?
+  isOldClaimProcess: boolean;
+  statementKind?: StatementKind;
 }
 
-function Claim ({ button, className, ethereumAddress }: Props): React.ReactElement<Props> | null {
+interface ConstructTx {
+  params?: any[];
+  tx?: string;
+}
+
+// Depending on isOldClaimProcess, construct the correct tx.
+function constructTx (accountId: string, ethereumSignature: string | null, kind: StatementKind | undefined, isOldClaimProcess: boolean): ConstructTx {
+  if (!ethereumSignature || !kind) {
+    return {};
+  }
+
+  return isOldClaimProcess
+    ? { params: [accountId, ethereumSignature], tx: 'claims.claim' }
+    : { params: [accountId, ethereumSignature, getStatement(kind)], tx: 'claims.claimAttest' };
```

Comment:
As an aside, we could just directly construct the tx here and use `const extrinsic = constructTx(...)` and then in TxButton use `extrinsic={extrinsic}` as opposed to the `tx` and `params`. (Which is also more efficient, since we don't need to re-do that stuff unless something changes) 

Here is an example as done for payouts

https://github.com/polkadot-js/apps/blob/master/packages/page-staking/src/Payouts/PayButton.tsx#L111

Nothing wrong here, just suggesting another option which may fit better for the use-case or thinking. 

(Longer term thinking of moving more stuff in that direction, since atm the construction params are effectively untyped and there are too many ways of doing the same thing)